N2A/src/gov/sandia/n2a/ui/settings/SettingsRepo.java

```diff
@@ -785,38 +769,52 @@ public void rebuild ()
     {
         if (! needRebuild) return;
 
-        List<MNode> modelContainers     = new ArrayList<MNode> ();
-        List<MNode> referenceContainers = new ArrayList<MNode> ();
+        Map<String, List<MNode>> existingContainers = new HashMap<String, List<MNode>> ();
         String primary = AppData.state.get ("Repos", "primary");
         for (MNode repo : repoModel.repos)
         {
             String repoName = repo.key ();
             boolean isPrimary = repoName.equals (primary);
             if (! repo.getBoolean ("visible")  &&  ! repo.getBoolean ("editable")  &&  ! isPrimary) continue;
 
-            MNode models     = getModels     (repoName);
-            MNode references = getReferences (repoName);
+            Map<String, MNode> dirs = new HashMap<String, MNode> ();
+            existing.forEach ((k,v) -> dirs.put(k, getExisting(k, repoName)));
             if (isPrimary)
             {
-                modelContainers    .add (0, models);
-                referenceContainers.add (0, references);
+                dirs.forEach ((k,v) -> {
+                    List<MNode> nodes = existingContainers.get (k);
+                    if(nodes == null) 
+                    {
+                        existingContainers.put (k, new ArrayList<MNode> ());
+                        nodes = existingContainers.get(k);
+                    }
+                    nodes.add (0, v);
+                });
             }
             else
             {
-                modelContainers    .add (models);
-                referenceContainers.add (references);
+                dirs.forEach ((k,v) -> {
+                    List<MNode> nodes = existingContainers.get (k);
+                    if(nodes == null) 
+                    {
+                        existingContainers.put (k, new ArrayList<MNode> ());
+                        nodes = existingContainers.get(k);
+                    }
+                    nodes.add (v);
+                });
             }
         }
-        AppData.models    .init (modelContainers);     // Triggers change() call to PanelModel
-        AppData.references.init (referenceContainers); // Triggers change() call to PanelReference
+        AppData.documents.forEach (c -> ((MCombo) c).init (existingContainers.get (c.key ()))); // Triggers change() call to PanelModel and PanelReference)
         needRebuild = false;
     }
 
     public boolean isEmpty (String key)
     {
-        MNode models     = getModels     (key);
-        MNode references = getReferences (key);
-        return  models.size () == 0  &&  references.size () == 0;
+        ArrayList<MNode> dirs = new ArrayList<> ();
+        existing.forEach ((k,v) -> dirs.add(getExisting(k, key)));
+        boolean empty = true;
+        for(MNode e: dirs) empty = empty && (e.size () == 0);
+        return empty;
```

Comment:
Iterate over existing. Retrieve the appropriate subdir. If it exists and is non-empty, return false. If flow of control reaches end of function, return true.